src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -2247,7 +2247,7 @@ private static void parseTownSetHomeblock(Player player, Town town, @Nullable Na
 				throw new TownyException(preEvent.getCancelMessage());
 			
 			// Test whether towns will be removed from the nation
-			if (nation != null && TownySettings.getNationRequiresProximity() > 0) {
+			if (nation != null && TownySettings.getNationRequiresProximity() > 0 && town.isCapital()) {

```

Comment:
This change would make it so that a town in the nation (which isn't the capital) could move their homeblock outside of the required range and not be kicked from the nation, as far as I can tell.

Reply:
The distance to the nation's capital gets checked in the setHomeblock method later on.